sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -341,6 +342,25 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
     }
   }
 
+  test("describe partition") {
+    withTable("partitioned_table") {
+      sql("CREATE TABLE partitioned_table (a STRING, b INT) PARTITIONED BY (c STRING, d STRING)")
+      sql("ALTER TABLE partitioned_table ADD PARTITION (c='Us', d=1)")
+
+      sql("DESC partitioned_table PARTITION (c='Us', d=1)")
```

Comment:
Hive supports describe based on the partition specified and will list the details for the particular partition when used with formatted or extended option. 
DESCRIBE formatted part_table partition (d='abc') 
https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-Describe

@dongjoon-hyun, this might be beyond the scope of this PR but this would be useful if there are a lot of partitions and we want to find details for a given partition.  What do you think? Thanks.